### DISCUSSION TITLE
Why there are difference in the results between Comsol and PyBaMM?

### DISCUSSION THREAD

**Discussion by ljt999** [pybamm-team/PyBaMM#3296] [Q&A] [answered]
Dear PyBaMM team,

First and foremost, I'd like to commend you for the exceptional work on PyBaMM. It runs both fast and conveniently, and I genuinely appreciate the efforts you've put into it.

I've come across a discrepancy that I hope you could give me some suggestions. Despite ensuring identical parameters (geometry, material properties, tolerance settings, etc.) in both Comsol and PyBaMM, I'm not getting consistent results. For your reference, I've configured PyBaMM to use the CasADi solver, the same one used by Comsol. I provide some comparisons of the results below:

![image](https://github.com/pybamm-team/PyBaMM/assets/83743280/1b516059-b1c5-4050-a41b-0e7c2d4d527e)
![image](https://github.com/pybamm-team/PyBaMM/assets/83743280/15ae4ba3-aa21-4ce3-a66c-5f2c6d79411c)
![image](https://github.com/pybamm-team/PyBaMM/assets/83743280/06c7bce4-e86a-4c06-bd48-bf22480bc961)

Based on these results, could you offer some insights or recommendations regarding the discrepancies observed? I truly value your time and expertise. Thank you in advance!

**Comment by brosaplanella** [comment]
Can you share a minimum working example of the PyBaMM code?

**Reply by ljt999** [reply]
Sure
-----------------------------------
import pybamm
import numpy as np
model = pybamm.lithium_ion.DFN()
var_pts = {
"x_n": 490, # negative electrode
"x_s": 73, # separator
"x_p": 437, # positive electrode
"r_n": 20, # negative particle
"r_p": 20, # positive particle
}
parameter_values = pybamm.ParameterValues("Chen2020")
solver = pybamm.CasadiSolver(mode="fast", atol=1e-10, rtol=1e-6)
sim = pybamm.Simulation(model,parameter_values=parameter_values,var_pts=var_pts,solver=solver)
sim.solve([0, 3600])
sim.plot()
-----------------------------------
As for other parameters, I amend them in Chen2020. I appreciate your time and assistence!

**Reply by brosaplanella** [reply]
It is certainly a parameter issue, so hard to tell where the problem is. My first guess would be to check the exchange current densities as they are often defined in different ways (see [this file](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/Chen2020.py) for the exact definition of the Chen 2020 parameters that PyBaMM uses).

**Reply by ljt999** [reply]
Thanks for your suggestion, I would recheck and test it.

**Reply by ljt999** [reply]
Hi Dr. Ferran Brosa Planella,

I've carefully matched the parameters with those in COMSOL and adjusted the exchange current densities to the same format used there. However, the more I adjust, the stranger the curves appear. I'm wondering if the Chen2020 model is highly sensitive to parameter changes. I've tweaked the OCV of the cathode and anode, the formula for exchange current densities, the electrolyte's diffusivity and conductivity, and the particle radius, among other things. Can you offer any advice on adjusting the parameters? I appreciate your time and patience in advance. 
Below are the modified Chen2020 parameters and the corresponding graph:

-----------------------------------------
```python3
import pybamm


def graphite_LGM50_ocp_Chen2020(sto):
    """
    LG M50 Graphite open-circuit potential as a function of stochiometry, fit taken
    from [1].

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    sto: :class:`pybamm.Symbol`
        Electrode stochiometry

    Returns
    -------
    :class:`pybamm.Symbol`
        Open-circuit potential
    """

    u_eq = (
       0.124 + 
       1.5 * pybamm.exp(- 70 * sto) - 
       0.0351 * pybamm.tanh((sto - 0.286) / 0.083) - 
       0.0045 * pybamm.tanh((sto - 0.9) / 0.119) - 
       0.035 * pybamm.tanh((sto - 0.99) / 0.05) - 
       0.0147 * pybamm.tanh((sto - 0.5) / 0.034) - 
       0.102 * pybamm.tanh((sto - 0.194) / 0.142) - 
       0.022 * pybamm.tanh((sto - 0.98) / 0.0164) - 
       0.011 * pybamm.tanh((sto - 0.124) / 0.0226) +
       0.0155 * pybamm.tanh((sto - 0.105) / 0.029)
    )


    return u_eq


def graphite_LGM50_electrolyte_exchange_current_density_Chen2020(
    c_e, c_s_surf, c_s_max, T
):
    """
    Exchange-current density for Butler-Volmer reactions between graphite and LiPF6 in
    EC:DMC.

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    c_e : :class:`pybamm.Symbol`
        Electrolyte concentration [mol.m-3]
    c_s_surf : :class:`pybamm.Symbol`
        Particle concentration [mol.m-3]
    c_s_max : :class:`pybamm.Symbol`
        Maximum particle concentration [mol.m-3]
    T : :class:`pybamm.Symbol`
        Temperature [K]

    Returns
    -------
    :class:`pybamm.Symbol`
        Exchange-current density [A.m-2]
    """
    k0_an = 3e-11 

    return (
        k0_an * c_e**0.5 * c_s_surf**0.5 * (c_s_max - c_s_surf) ** 0.5
    )



def nmc_LGM50_ocp_Chen2020(sto):
    """
    LG M50 NMC open-circuit potential as a function of stochiometry, fit taken
    from [1].

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    sto: :class:`pybamm.Symbol`
        Electrode stochiometry

    Returns
    -------
    :class:`pybamm.Symbol`
        Open-circuit potential
    """

    u_eq = (
        -63.73349423*(1.062*sto)**5
        + 174.82930386*(1.062*sto)**4
        - 180.52045189*(1.062*sto)**3
        + 88.86774398*(1.062*sto)**2
        - 23.12236785*(1.062*sto)
        + 7.07949884
    )

    return u_eq


def nmc_LGM50_electrolyte_exchange_current_density_Chen2020(c_e, c_s_surf, c_s_max, T):
    """
    Exchange-current density for Butler-Volmer reactions between NMC and LiPF6 in
    EC:DMC.

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    c_e : :class:`pybamm.Symbol`
        Electrolyte concentration [mol.m-3]
    c_s_surf : :class:`pybamm.Symbol`
        Particle concentration [mol.m-3]
    c_s_max : :class:`pybamm.Symbol`
        Maximum particle concentration [mol.m-3]
    T : :class:`pybamm.Symbol`
        Temperature [K]

    Returns
    -------
    :class:`pybamm.Symbol`
        Exchange-current density [A.m-2]
    """
    k0_cat = 3e-11  
    return (
        k0_cat * c_e**0.5 * c_s_surf**0.5 * (c_s_max - c_s_surf) ** 0.5
    )



def electrolyte_diffusivity_Nyman2008(c_e, T):
    """
    Diffusivity of LiPF6 in EC:EMC (3:7) as a function of ion concentration. The data
    comes from [1]

    References
    ----------
    .. [1] A. Nyman, M. Behm, and G. Lindbergh, "Electrochemical characterisation and
    modelling of the mass transport phenomena in LiPF6-EC-EMC electrolyte,"
    Electrochim. Acta, vol. 53, no. 22, pp. 6356–6365, 2008.

    Parameters
    ----------
    c_e: :class:`pybamm.Symbol`
        Dimensional electrolyte concentration
    T: :class:`pybamm.Symbol`
        Dimensional temperature

    Returns
    -------
    :class:`pybamm.Symbol`
        Solid diffusivity
    """

    D_c_e = 1e-4 * 10 ** (-4.43 - 54 / (T - (229 + 5 * (c_e / 1000))) - 0.22 * (c_e / 1000))


    # Nyman et al. (2008) does not provide temperature dependence

    return D_c_e


def electrolyte_conductivity_Nyman2008(c_e, T):
    """
    Conductivity of LiPF6 in EC:EMC (3:7) as a function of ion concentration. The data
    comes from [1].

    References
    ----------
    .. [1] A. Nyman, M. Behm, and G. Lindbergh, "Electrochemical characterisation and
    modelling of the mass transport phenomena in LiPF6-EC-EMC electrolyte,"
    Electrochim. Acta, vol. 53, no. 22, pp. 6356–6365, 2008.

    Parameters
    ----------
    c_e: :class:`pybamm.Symbol`
        Dimensional electrolyte concentration
    T: :class:`pybamm.Symbol`
        Dimensional temperature

    Returns
    -------
    :class:`pybamm.Symbol`
        Solid diffusivity
    """

    sigma_e = (
       1 / 10 * (c_e / 1000) * (-10.5 + 0.668 * (c_e /1000) + 0.494 * (c_e /1000) ** 2 + 0.0740 * T - 0.0178 * (c_e / 1000) * T - 8.86e-4 * (c_e /1000) ** 2 * T - 6.96e-5 * T ** 2 + 2.8e-5 * (c_e /1000) * T ** 2) ** 2
    )

    # Nyman et al. (2008) does not provide temperature dependence

    return sigma_e


# Call dict via a function to avoid errors when editing in place
def get_parameter_values():
    """
    Parameters for an LG M50 cell, from the paper

        Chang-Hui Chen, Ferran Brosa Planella, Kieran O'Regan, Dominika Gastol, W.
        Dhammika Widanage, and Emma Kendrick. Development of Experimental Techniques for
        Parameterization of Multi-scale Lithium-ion Battery Models. Journal of The
        Electrochemical Society, 167(8):080534, 2020. doi:10.1149/1945-7111/ab9050.

    and references therein.

    SEI parameters are example parameters for SEI growth from the papers:

        Ramadass, P., Haran, B., Gomadam, P. M., White, R., & Popov, B. N. (2004).
        Development of first principles capacity fade model for Li-ion cells. Journal of
        the Electrochemical Society, 151(2), A196-A203.

        Ploehn, H. J., Ramadass, P., & White, R. E. (2004). Solvent diffusion model for
        aging of lithium-ion battery cells. Journal of The Electrochemical Society,
        151(3), A456-A462.

        Single, F., Latz, A., & Horstmann, B. (2018). Identifying the mechanism of
        continued growth of the solid-electrolyte interphase. ChemSusChem, 11(12),
        1950-1955.

        Safari, M., Morcrette, M., Teyssot, A., & Delacour, C. (2009). Multimodal
        Physics- Based Aging Model for Life Prediction of Li-Ion Batteries. Journal of
        The Electrochemical Society, 156(3),

        Yang, X., Leng, Y., Zhang, G., Ge, S., Wang, C. (2017). Modeling of lithium
        plating induced aging of lithium-ion batteries: Transition from linear to
        nonlinear aging. Journal of Power Sources, 360, 28-40.

    Note: this parameter set does not claim to be representative of the true parameter
    values. Instead these are parameter values that were used to fit SEI models to
    observed experimental data in the referenced papers.
    """

    return {
        "chemistry": "lithium_ion",
        # sei
        "Ratio of lithium moles to SEI moles": 2.0,
        "Inner SEI reaction proportion": 0.5,
        "Inner SEI partial molar volume [m3.mol-1]": 9.585e-05,
        "Outer SEI partial molar volume [m3.mol-1]": 9.585e-05,
        "SEI reaction exchange current density [A.m-2]": 1.5e-07,
        "SEI resistivity [Ohm.m]": 200000.0,
        "Outer SEI solvent diffusivity [m2.s-1]": 2.5000000000000002e-22,
        "Bulk solvent concentration [mol.m-3]": 2636.0,
        "Inner SEI open-circuit potential [V]": 0.1,
        "Outer SEI open-circuit potential [V]": 0.8,
        "Inner SEI electron conductivity [S.m-1]": 8.95e-14,
        "Inner SEI lithium interstitial diffusivity [m2.s-1]": 1e-20,
        "Lithium interstitial reference concentration [mol.m-3]": 15.0,
        "Initial inner SEI thickness [m]": 2.5e-09,
        "Initial outer SEI thickness [m]": 2.5e-09,
        "EC initial concentration in electrolyte [mol.m-3]": 4541.0,
        "EC diffusivity [m2.s-1]": 2e-18,
        "SEI kinetic rate constant [m.s-1]": 1e-12,
        "SEI open-circuit potential [V]": 0.4,
        "SEI growth activation energy [J.mol-1]": 0.0,
        "Negative electrode reaction-driven LAM factor [m3.mol-1]": 0.0,
        "Positive electrode reaction-driven LAM factor [m3.mol-1]": 0.0,
        # cell
        "Negative current collector thickness [m]": 1.2e-05,
        "Negative electrode thickness [m]": 1.68e-04,
        "Separator thickness [m]": 2.5e-05,
        "Positive electrode thickness [m]": 1.5e-05,
        "Positive current collector thickness [m]": 1.6e-05,
        "Electrode height [m]": 0.065,
        "Electrode width [m]": 1.58,
        "Cell cooling surface area [m2]": 0.00531,
        "Cell volume [m3]": 2.42e-05,
        "Cell thermal expansion coefficient [m.K-1]": 1.1e-06,
        "Negative current collector conductivity [S.m-1]": 58411000.0,
        "Positive current collector conductivity [S.m-1]": 36914000.0,
        "Negative current collector density [kg.m-3]": 8960.0,
        "Positive current collector density [kg.m-3]": 2700.0,
        "Negative current collector specific heat capacity [J.kg-1.K-1]": 385.0,
        "Positive current collector specific heat capacity [J.kg-1.K-1]": 897.0,
        "Negative current collector thermal conductivity [W.m-1.K-1]": 401.0,
        "Positive current collector thermal conductivity [W.m-1.K-1]": 237.0,
        "Nominal cell capacity [A.h]": 5.0,
        "Current function [A]": 5.0,
        "Contact resistance [Ohm]": 0,
        # negative electrode
        "Negative electrode conductivity [S.m-1]": 100.0,
        "Maximum concentration in negative electrode [mol.m-3]": 31507.0,
        "Negative electrode diffusivity [m2.s-1]": 9e-14,
        "Negative electrode OCP [V]": graphite_LGM50_ocp_Chen2020,
        "Negative electrode porosity": 0.33,
        "Negative electrode active material volume fraction": 0.67,
        "Negative particle radius [m]": 1e-06,
        "Negative electrode Bruggeman coefficient (electrolyte)": 1.5,
        "Negative electrode Bruggeman coefficient (electrode)": 1.5,
        "Negative electrode charge transfer coefficient": 0.5,
        "Negative electrode double-layer capacity [F.m-2]": 0.2,
        "Negative electrode exchange-current density [A.m-2]"
        "": graphite_LGM50_electrolyte_exchange_current_density_Chen2020,
        "Negative electrode density [kg.m-3]": 1657.0,
        "Negative electrode specific heat capacity [J.kg-1.K-1]": 700.0,
        "Negative electrode thermal conductivity [W.m-1.K-1]": 1.7,
        "Negative electrode OCP entropic change [V.K-1]": 0.0,
        # positive electrode
        "Positive electrode conductivity [S.m-1]": 10,
        "Maximum concentration in positive electrode [mol.m-3]": 49761.0,
        "Positive electrode diffusivity [m2.s-1]": 1e-14,
        "Positive electrode OCP [V]": nmc_LGM50_ocp_Chen2020,
        "Positive electrode porosity": 0.25,
        "Positive electrode active material volume fraction": 0.75,
        "Positive particle radius [m]": 1e-06,
        "Positive electrode Bruggeman coefficient (electrolyte)": 1.5,
        "Positive electrode Bruggeman coefficient (electrode)": 1.5,
        "Positive electrode charge transfer coefficient": 0.5,
        "Positive electrode double-layer capacity [F.m-2]": 0.2,
        "Positive electrode exchange-current density [A.m-2]"
        "": nmc_LGM50_electrolyte_exchange_current_density_Chen2020,
        "Positive electrode density [kg.m-3]": 3262.0,
        "Positive electrode specific heat capacity [J.kg-1.K-1]": 700.0,
        "Positive electrode thermal conductivity [W.m-1.K-1]": 2.1,
        "Positive electrode OCP entropic change [V.K-1]": 0.0,
        # separator
        "Separator porosity": 0.55,
        "Separator Bruggeman coefficient (electrolyte)": 1.5,
        "Separator density [kg.m-3]": 397.0,
        "Separator specific heat capacity [J.kg-1.K-1]": 700.0,
        "Separator thermal conductivity [W.m-1.K-1]": 0.16,
        # electrolyte
        "Initial concentration in electrolyte [mol.m-3]": 1000.0,
        "Cation transference number": 0.39,
        "Thermodynamic factor": 1.0,
        "Electrolyte diffusivity [m2.s-1]": electrolyte_diffusivity_Nyman2008,
        "Electrolyte conductivity [S.m-1]": electrolyte_conductivity_Nyman2008,
        # experiment
        "Reference temperature [K]": 298.15,
        "Total heat transfer coefficient [W.m-2.K-1]": 10.0,
        "Ambient temperature [K]": 298.15,
        "Number of electrodes connected in parallel to make a cell": 1.0,
        "Number of cells connected in series to make a battery": 1.0,
        "Lower voltage cut-off [V]": 2.7,
        "Upper voltage cut-off [V]": 4.2,
        "Initial concentration in negative electrode [mol.m-3]": 27411.0,
        "Initial concentration in positive electrode [mol.m-3]": 22392.0,
        "Initial temperature [K]": 298.15,
        # citations
        "citations": ["Chen2020"],
    }
```
---------------------------------------------------

    
![comsol_pybamm_1c](https://github.com/pybamm-team/PyBaMM/assets/83743280/355b9d59-5607-462a-81c2-b5c39851ceff)

**Reply by brosaplanella** [reply]
What is sensitive to parameters is the model (e.g. SPM, DFN...), not the parameter set (e.g. Chen 2020), so it is a matter to find if any parameters are defined slightly different in COMSOL than in PyBaMM, but in theory they should be the exact same set. I suspect the OCVs are defined the same way in COMSOL and in PyBaMM (the only difference might be whether the variable is concentration or stoichiometry), so changing the fits will not give a better agreement.

Is there any way to find which equations is COMSOL exactly solving for (as in, the exact mathematical formulation of the equations)? That would help check discrepancies with the PyBaMM model.

**Reply by ljt999** [reply]
Your assistance has been immensely helpful to me; I've resolved the issue. I greatly appreciate your help! Thank you so much!

**Reply by brosaplanella** [reply]
No worries! May I ask what the issue was, so we know it in case someone encounters it again in the future?

**Reply by ljt999** [reply, marked as answer]
I realized that the definitions of exchange current factors differ between COMSOL and PyBaMM. In COMSOL, I defined it using flux, so I need to multiply by the Faraday constant in PyBaMM. I appreciate your assistance again!

**Reply by Brushnie33** [reply]
Hello, I was wondering if there is a link for the comsol model that is used for comparison against the pybamm DFN model (these in this folder: https://github.com/pybamm-team/PyBaMM/tree/977dcf9cdfc2fc978da3b1b56b7423454a2a2929/examples/scripts/compare_comsol) shows perfect match. But when I used Okane2022 parameters for comparison, and I also converted reaction rate constant (multiply by 1000^0.5 and divided by Faraday's constant, 1000 is the electrolyte reference concentration). The results show good match for C-rates up to 0.50C, but there is discrepancy when C-rate goes above 1C. What could be the cause for that? @brosaplanella 
![image](https://github.com/user-attachments/assets/63283804-c9c4-448b-b1f0-7b7faca6d3d7)

**Reply by brosaplanella** [reply]
The simulation was done a few years ago by @rtimms, I believe. Not sure if he still has the file he used for validation. The fact that agreement is good at low C-rates and gets work at higher C-rates suggests some of the parameters related to the dynamics not being correct, so I would double check that. I am also a bit confused on why the curves stop at different voltages. Is it a resolution issue?

**Reply by Brushnie33** [reply]
@brosaplanella @rtimms  I was suspecting that this difference might be coming from different stoichiometric window, in COMSOL, users specify minimum and maximum SOCs for each electrode, which is not explicitly specified in PyBaMM. Please neglect the stopping voltage (I didn't specify the stop voltage in COMSOL so the simulation was more time-based, but I can add that to match). I also did a comparison using half-cell set-up, the results are much better than the full-cell case above, but still see some differences. 
![image](https://github.com/user-attachments/assets/e48adfb7-5fdb-43a6-8c77-88adf30a4867)

**Reply by rtimms** [reply]
I don't have access to that file anymore, but it was for the Marquis 2019 parameters. If you're using a different parameters set and the half-cells fit well but not the full cell then it is probably the electrode balance. 

You can get the stoichiometry windows by doing 
```python3
parameter_values = pybamm.ParameterValues("OKane2022")
pybamm.lithium_ion.get_min_max_stoichiometries(parameter_values)
```

We don't set min/max sto directly, but instead calculate it to be consistent with the OCPs, electrode capacities, and voltage limits.

**Reply by Sihui-Chen** [reply]
> I realized that the definitions of exchange current factors differ between COMSOL and PyBaMM. In COMSOL, I defined it using flux, so I need to multiply by the Faraday constant in PyBaMM. I appreciate your assistance again!

Thank you for your valuable experience. I also suffer from the problem of discharge voltage discrepancy. I make sure the geometry parameters, solid-state parameters, liquid-state parameters, initial and maximum concentration in both electrodes, and OCP function settings in COMSOL the same as PyBaMM (Chen2020). And I set the exchange current density in COMSOL the same as PyBaMM (Chen2020), but the voltage mismatch still exists. 

In COMSOL, I choose Butlet-Volmer equation for local current density calculation, but in PyBamm I failed to find any specific definition of local current density. I don't know if the difference of local current definition cause the voltage discrepancy or other factors. 


<img width="587" height="417" alt="image" src="https://github.com/user-attachments/assets/a026c6c9-6280-43b8-9cad-ed1e1d1267fd" />
<img width="696" height="304" alt="image" src="https://github.com/user-attachments/assets/3344207d-096d-48b3-9ca7-d00d1bc7f944" />

**Comment by Brushnie33** [comment]
@rtimms I used esoh function to calculate the stoichiometries, and the results still didn't match well. I also find out that when I increase the thickness (from 4.2e-5(Xu2019) to 7.56e-5(Okane2022), about 80% increase) of the positive electrode in the half-cell (also scaled current/capacity, but all other parameters remain the same), then the difference between the half-cell cell voltage in pybamm and comsol increases substantially. So the problem is not likely due to the stoichiometries, but maybe mesh or solver issues. Any thoughts? 
![image](https://github.com/user-attachments/assets/60bc7d1b-4040-48ce-b8cb-2c3301873143)

**Reply by rtimms** [reply]
It’s quite hard to say without knowing more details of the implementation and parameters used. One thing to try is to simulate different limiting cases to try and debug the problem (eg low C-rate looks good so capacities and OCPs are probably fine, but things go wrong at higher rate). Try to probe whether this is kinetics, transport etc.

**Reply by Brushnie33** [reply]
@rtimms After further investigation, I have found that the issue comes from the electrolyte conductivity and diffusivity, it might be the units of c_e in comsol and pybamm are different. When I used constant values for electrolyte conductivity and diffusivity, the results match well for both half-cell and full-cell configurations. If I used an analytical equation or interpolation, there is a big mismatch for higher C rates.

**Comment by Sihui-Chen** [comment]
Dear PyBaMM team,

I am currently trying to reproduce the PyBaMM SPM default simulation results in COMSOL. However, the voltage curves do not match well, as shown in Figure 1.

I looked at the example “Comparison of PyBaMM and COMSOL Discharge Curves”, where the COMSOL and PyBaMM voltage curves overlap very well. This is exactly what I hope to achieve.

I have already adjusted the geometry parameters, initial solid-phase concentration, maximum solid-phase concentration, and other parameters, but the voltage curves still do not align properly.

Could you please let me know if the original COMSOL model used to generate the simulation data in “Comparison of PyBaMM and COMSOL Discharge Curves” is still available? I would like to study it and check what I might have missed in my adjustments.

Thank you very much for your help!

<img width="587" height="417" alt="image" src="https://github.com/user-attachments/assets/0e10af0e-9348-4eca-9ae6-26857431d262" />

Best,
Sihui
